bamshuf.c

```diff
@@ -573,7 +587,20 @@ char * generate_prefix() {
         perror("collate");
         return NULL;
     }
-    snprintf(prefix, PREFIX_LEN, "/tmp/collate%x", pid);
+
+    int use_default = 1;
+    char *tmp_env = getenv("TMPDIR");
```

Comment:
On my Mac, `TMPDIR` is 49 characters long.  It does fit in PREFIX_LEN, but only just.  Possibly it would be better to measure the length of `TMPDIR` and then allocate a sufficiently long buffer?  Plus, it would make the no TMPDIR easier to handle, by just doing:
```
    if (!tmp_env)
        tmp_env = "/tmp";
    size_t prefix_len = strlen(tmp_env) + 16;
    // ...
```

Reply:
Agreed, although what on earth is the Mac doing with such a rediculously long temp dir?

Most of that old cold was in earlier commits, but I've pushed a newer one that can cope with uint32_t max length pid (they may get that high on some clusters perhaps).  No need to check for snprintf return as it can no longer overflow, and even if it does it'll have enough bytes of random prefix by then.